waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/MetaStoreMappingImpl.java

```diff
@@ -53,6 +56,9 @@ public String transformOutboundDatabaseName(String databaseName) {
 
   @Override
   public ThriftHiveMetastore.Iface getClient() {
```

Comment:
Are calls to this single threaded?

Reply:
yes, nothing has changed here. Request comes in a MetastoreMapping is being made and the request is handled by the client, nothing asynchronous going on here.

On an aside. This is bad method that's leaking the client. Not trivial to get rid of it though.